v2/metadata/client.go

```diff
@@ -0,0 +1,365 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metadata contains types to record/retrieve metadata stored in MLMD
+// for individual pipeline steps.
+package metadata
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/golang/glog"
+	pb "github.com/kubeflow/pipelines/v2/third_party/ml_metadata"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	pipelineContextTypeName    = "kfp.Pipeline"
+	pipelineRunContextTypeName = "kfp.PipelineRun"
+	containerExecutionTypeName = "kfp.ContainerExecution"
+)
+
+var (
+	// Note: All types are schemaless so we can easily evolve the types as needed.
+	pipelineContextType = &pb.ContextType{
+		Name: proto.String(pipelineContextTypeName),
+	}
+
+	pipelineRunContextType = &pb.ContextType{
+		Name: proto.String(pipelineRunContextTypeName),
+	}
+
+	containerExecutionType = &pb.ExecutionType{
+		Name: proto.String(containerExecutionTypeName),
+	}
+)
+
+// Client is an MLMD service client.
+type Client struct {
+	svc pb.MetadataStoreServiceClient
+}
+
+// NewClient creates a Client given the MLMD server address and port.
+func NewClient(serverAddress, serverPort string) (*Client, error) {
+	conn, err := grpc.Dial(fmt.Sprintf("%s:%s", serverAddress, serverPort), grpc.WithInsecure())
+	if err != nil {
+		return nil, err
```

Comment:
For errors returned from libraries, I believe it's always good practice to wrap it via either:

* https://blog.golang.org/go1.13-errors#TOC_3.3.
* or https://github.com/pkg/errors

Because if we do not wrap the error, there's no call stack trace, so it would be hard to find out where an error occurred when we actually see it.

Reply:
Yep, agreed with you. I did not make this change since the PR was getting big. I'm going to leave a TODO here for this as well.